packages/slate/src/interfaces/scrubber.ts

```diff
@@ -0,0 +1,65 @@
+export type Scrubber = (key: string, value: unknown) => unknown
+
+export interface ScrubberInterface {
+  setScrubber(scrubber: Scrubber | undefined): void
+  stringify(value: any): string
+  textRandomizer(fieldNames: string[]): Scrubber
+}
+
+let _scrubber: Scrubber | undefined = undefined
+
+/**
+ * This interface implements a stringify() function, which is used by Slate
+ * internally when generating exceptions containing end user data. Developers
+ * using Slate may call Scrubber.setScrubber() to alter the behavior of this
+ * stringify() function.
+ *
+ * For example, to prevent the cleartext logging of 'text' fields within Nodes:
+ *
+ *    import { Scrubber } from 'slate';
+ *    Scrubber.setScrubber((key, val) => {
+ *      if (key === 'text') return '...scrubbed...'
+ *      return val
+ *    });
+ *
+ */
+export const Scrubber: ScrubberInterface = {
+  setScrubber(scrubber: Scrubber | undefined): void {
+    _scrubber = scrubber
+  },
+
+  stringify(value: any): string {
+    return JSON.stringify(value, _scrubber)
+  },
+
+  /**
+   * Helper function to create a Scrubber that scrubs the contents of any field
+   * in `fieldNames` with a randomly generated string of the same length.
+   *
+   * For example:
+   *
+   *    Scrubber.setScrubber(Scrubber.textRandomizer(['text']));
+   */
+  textRandomizer(fieldNames: string[]): Scrubber {
```

Comment:
While I appreciate the addition of a helper function, the way it's implemented as an object property it can't be tree shaken when unused.

I don't think you need the `ScrubberInterface` and this sub-object at all; if you unwrap it and just `export const` the three functions inside it should be an equivalent interface.

Reply:
We thought we'd keep in line with how other Slate modules look (e.g., `Transforms.foo`, `Editor.bar`) so we could get `Scrubber.setScrubber`. Also made documentation a bit easier as we could just add a "Scrubber" section to the docs.

But we're fine with exporting them at the root level too, so you get something like:
```
import { Editor, Transforms, Node, Element, setScrubber, textRandomizer createEditor  } from 'slate'
```

Reply:
Thanks for the feedback! I've updated the branch to remove `textRandomizer` from the code, since it was just a convenience function anyway. I've instead included it as an example in the `docs/api/scrubber.md` documentation page.

Reply:
I guess I'm used to using `import * as Scrubber` internally and forgot Slate was structured this way. As this is an internal API, however, it could still be used by integrators as `import { Scrubber } from 'slate'` with a change to how `index.ts` is structured.

Moving it to documentation is a good solution. Maybe I'll look at this later as a wider change to improve tree shaking within Slate.